Request - Do not expose "check" identifier

##### System information (version)

- OpenCV => 4.3
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2019



##### Detailed description

<!-- your description -->
OpenCV cannot be included in a UE4(Unreal Engined) project. (Tested with UE4 4.24)
This is due to the check() functions in modules/core/include/opencv2/core/utility.hpp and in modules/core/include/opencv2/core/cvstd_wrapper.hpp and several other files.
UE4 defines the macro check() to be used for assertions. This is deeply integrated into their workflow and probably cannot be changed.
The macro collides with the check() functions.
According to a warning generated on line 52 of utility.hpp, the same issue occurs when using Apple headers.

As this is an issue that now appears when using 2 different major codebases, I suggest avoiding exposing the "check" identifier.
For the existing functions, e.g. in utility.hpp and in cvstd_wrapper.hpp we can alter the name to cv_check().

##### Steps to reproduce
Include OpenCV 4.3.0 in a UE4 4.24 game and try building.
An example of one of the errors that will be generated:
ThirdParty\opencv\include\opencv2\core\cvstd_wrapper.hpp(45): error C4002: too many arguments for function-like macro invocation 'check'

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   <!--
   OpenCV team works with answers.opencv.org, Stack Overflow and other communities
   to discuss problems. Tickets with question without real issue statement will be
   closed.
   -->
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   <!--
   Places to check:
   * OpenCV documentation: https://docs.opencv.org
   * FAQ page: https://github.com/opencv/opencv/wiki/FAQ
   * OpenCV forum: https://answers.opencv.org
   * OpenCV issue tracker: https://github.com/opencv/opencv/issues?q=is%3Aissue
   * Stack Overflow branch: https://stackoverflow.com/questions/tagged/opencv
   -->
 - [x] I updated to latest OpenCV version and the issue is still there
   <!--
   master branch for OpenCV 4.x and 3.4 branch for OpenCV 3.x releases.
   OpenCV team supports only latest release for each branch.
   The ticket is closed, if the problem is not reproduced with modern version.
   -->
 - [x] There is reproducer code and related data files: videos, images, onnx, etc
   <!--
   The best reproducer -- test case for OpenCV that we can add to the library.
   Recommendations for media files and binary files:
   * Try to reproduce the issue with images and videos in opencv_extra repository
     to reduce attachment size
   * Use PNG for images, if you report some CV related bug, but not image reader
     issue
   * Attach the image as archite to the ticket, if you report some reader issue.
     Image hosting services compress images and it breaks the repro code.
   * Provide ONNX file for some public model or ONNX file with with random weights,
     if you report ONNX parsing or handling issue. Architecture details diagram
     from netron tool can be very useful too. See https://lutzroeder.github.io/netron/
   -->

--- a/modules/core/include/opencv2/core/cvstd_wrapper.hpp
+++ b/modules/core/include/opencv2/core/cvstd_wrapper.hpp
@@ -42,11 +42,11 @@ struct has_parenthesis_operator
 {
 private:
     template<typename T>
-    static CV_CONSTEXPR std::true_type check(typename std::is_same<typename std::decay<decltype(std::declval<T>().operator()(std::declval<Args>()...))>::type, Ret>::type*);
+    static CV_CONSTEXPR std::true_type has_parenthesis_operator_check(typename std::is_same<typename std::decay<decltype(std::declval<T>().operator()(std::declval<Args>()...))>::type, Ret>::type*);
 
-    template<typename> static CV_CONSTEXPR std::false_type check(...);
+    template<typename> static CV_CONSTEXPR std::false_type has_parenthesis_operator_check(...);
 
-    typedef decltype(check<C>(0)) type;
+    typedef decltype(has_parenthesis_operator_check<C>(0)) type;
 
 public:
 #if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900/*MSVS 2015*/)

